contracts/FlashLoanV1Router02.sol

```diff
@@ -0,0 +1,200 @@
+            IFlashLoanV1Factory(factory).createPool(token);
+        }
+        address pool = FlashLoanV1Library.poolFor(factory, token);
+        TransferHelper.safeTransferFrom(token, msg.sender, pool, amount);
+        liquidity = IFlashLoanV1Pool(pool).mint(to);
+    }
+    function addLiquidityETH(
+        address to,
+        uint deadline
+    ) external virtual override payable ensure(deadline) returns (uint liquidity) {
+        if (IFlashLoanV1Factory(factory).getPool(WETH) == address(0)) {
+            IFlashLoanV1Factory(factory).createPool(WETH);
+        }
+        address pool = FlashLoanV1Library.poolFor(factory, WETH);
+        IWETH(WETH).deposit{value: msg.value}();
+        assert(IWETH(WETH).transfer(pool, msg.value));
+        liquidity = IFlashLoanV1Pool(pool).mint(to);
+    }
+
+    // **** REMOVE LIQUIDITY ****
+    function removeLiquidity(
+        address token,
+        uint liquidity,
+        address to,
+        uint deadline
+    ) public virtual override ensure(deadline) returns (uint amount) {
+        address pool = FlashLoanV1Library.poolFor(factory, token);
+        IFlashLoanV1Pool(pool).transferFrom(msg.sender, pool, liquidity);
+        amount = IFlashLoanV1Pool(pool).burn(to);
+    }
+    function removeLiquidityETH(
+        uint liquidity,
+        address to,
+        uint deadline
+    ) public virtual override ensure(deadline) returns (uint amountETH) {
+        amountETH = removeLiquidity(WETH, liquidity, address(this), deadline);
+        IWETH(WETH).withdraw(amountETH);
+        TransferHelper.safeTransferETH(to, amountETH);
+    }
+    function removeLiquidityWithPermit(
+        address token,
+        uint liquidity,
+        address to,
+        uint deadline,
+        bool approveMax, uint8 v, bytes32 r, bytes32 s
+    ) external virtual override returns (uint amount) {
+        address pool = FlashLoanV1Library.poolFor(factory, token);
+        uint value = approveMax ? uint(-1) : liquidity;
+        IFlashLoanV1Pool(pool).permit(msg.sender, address(this), value, deadline, v, r, s);
+        amount = removeLiquidity(token, liquidity, to, deadline);
+    }
+    function removeLiquidityETHWithPermit(
+        uint liquidity,
+        address to,
+        uint deadline,
+        bool approveMax, uint8 v, bytes32 r, bytes32 s
+    ) external virtual override returns (uint amountETH) {
+        address pool = FlashLoanV1Library.poolFor(factory, WETH);
+        uint value = approveMax ? uint(-1) : liquidity;
+        IFlashLoanV1Pool(pool).permit(msg.sender, address(this), value, deadline, v, r, s);
+        amountETH = removeLiquidityETH(liquidity, to, deadline);
+    }
+
+    /**
+     * @dev Get the Deerfi Pool that will be used as the source of a loan.
+     * @param token The loan currency.
+     * @return The Deerfi V2 Pool that will be used as the source of the flash loan.
+     */
+    function getPoolAddress(address token) public view returns (address) {
```

Comment:
inline this method. we don't need to have a `one line only` function.

Reply:
Done.